tests/components/netatmo/test_data_handler.py

```diff
@@ -0,0 +1,103 @@
+"""The tests for Netatmo data handler."""
+from time import time
+from unittest.mock import patch
+
+from homeassistant.components.netatmo import DOMAIN
+from homeassistant.components.netatmo.data_handler import (
+    CAMERA_DATA_CLASS_NAME,
+    NetatmoDataHandler,
+)
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+
+async def test_data_handler(hass, config_entry):
+    """Test the data handler."""
+    await hass.async_block_till_done()
+
+    with patch(
+        "homeassistant.components.netatmo.api.ConfigEntryNetatmoAuth"
+    ) as mock_auth:
+        hass.data[DOMAIN] = {config_entry.entry_id: {"netatmo_auth": mock_auth()}}
+
+    assert mock_auth.called
```

Comment:
Is it possible to avoid accessing `hass.data` here and patch the library during setup instead?

Reply:
Probably, but it might affect coverage. I'll take a look.

Reply:
Right now I don't see how to work around that without adding too much mocking/complexity.